src/containers/GetAnArticle.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import AnArticle from '../components/AnArticle';
+import Comments from '../components/Comments';
+import {
+  fetchAnArticle,
+  fetchArticleComments,
+} from '../store/actions/commentsActions';
+
+export class GetAnArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.slug = '';
+  }
+
+  componentDidMount() {
+    const { match } = this.props;
+    this.slug = match.params;
+    const { fetchAnArticle } = this.props;
+    const { fetchArticleComments } = this.props;
+    fetchAnArticle(this.slug);
+    fetchArticleComments(this.slug);
+  }
+
+  render() {
+    const articleRender = () => {
+      const { isFetchingArticle } = this.props;
+      const { errors } = this.props;
+      const { isFetchingComments } = this.props;
+      if (isFetchingArticle) {
+        return (<div className="article-loading">Loading...</div>);
+      }
+      if (errors) {
+        return (<div className="article-errors">Return 404...</div>);
+      }
+      if (isFetchingComments) {
+        return (<div className="comments-loading">Loading...</div>);
+      }
+      const { article } = this.props;
+      const { comments } = this.props;
+
+      return (
+        <div className="container an-article">
+          <div className="auth">
+            <AnArticle data={article} />
+            <br />
+            <hr />
+            <br />
+            <Comments data={comments} />
+          </div>
+        </div>
+      );
+    };
+
+    return (
+      <div>
+        {articleRender()}
+      </div>
+    );
```

Comment:
@Etomovich, make one return statement, this is a very bad approach to creating components.

Reply:
Okay let me refactor this piece of code.